pkg/vm/engine/tae/containers/cn_vectorwindow_adapter.go

```diff
@@ -119,81 +186,31 @@ func (win *vectorWindow[T]) Equals(o Vector) bool {
 	return true
 
 }
-
-func (win *vectorWindow[T]) GetView() VectorView {
-	return &vectorWindow[T]{
+func (win *CnTaeVectorWindow[T]) Window(offset, length int) Vector {
+	if offset+length > win.length {
+		panic("bad window param")
+	}
+	return &CnTaeVectorWindow[T]{
 		ref: win.ref,
 		windowBase: &windowBase{
-			offset: win.offset,
-			length: win.length,
+			offset: offset + win.offset,
+			length: length,
 		},
 	}
 }
 
-func (win *vectorWindow[T]) CloneWindow(offset, length int, allocator ...*mpool.MPool) Vector {
-	return win.ref.CloneWindow(offset+win.offset, length, allocator...)
+func (win *CnTaeVectorWindow[T]) WriteTo(w io.Writer) (int64, error) {
+	panic("not implemented")
 }
 
-func (win *vectorWindow[T]) Data() []byte {
-	return win.ref.DataWindow(win.offset, win.length)
+func (win *CnTaeVectorWindow[T]) GetView() VectorView {
```

Comment:
I think you can remove all not used APIs.

Reply:
sure.